HomeWork_12/HW12.py

```diff
@@ -0,0 +1,73 @@
+import json
+from datetime import datetime
+def timer():
+    start = datetime()
+    def inner():
+        return datetime() - start
+    return inner
+
+def funcname(func):
+    def inner(*args,**kwargs):
+        print(f"Функція {func.__name__}")
+        return func(*args,**kwargs)
+
+
+
+
+
+# phonebook ={
+#         "Aleks": "347638756",
+#         "Fedor": "3784638754",
+#         "Den": "2387463745"
+#
+#      }
+# json_phonebook = json.dumps(phonebook)
+
+try:
+    with open("/Users/AleksandrPitelin/PythonProject/PythonProject/HomeWork_12/Json_hw12", 'r') as file:
+        data = file.read()
+        phonebook = json.loads(data)
+except FileNotFoundError:
+    phonebook = {}
+
+
+while True:
+    # @timer(command)
+    # @funcname(command)
+    command = input('Enter a command: ')
+    if command == 'list':
+        for key in phonebook:
+            print(key)
+    elif command == 'stats':
+        L = phonebook
+        print(len(L))
+    elif command == 'show':
+        s = input("Enter a name: ")
+        if phonebook.get(s):
+            print(phonebook[s])
+    elif command == 'add':
+        n = input("Enter a name: ")
+        p = int(input("Enter a phone: "))
+
+        # if n != phonebook:
+        #     phonebook[n] = p
+        if n not in phonebook:
+            phonebook[n] = p
+            with open("/Users/AleksandrPitelin/PythonProject/PythonProject/HomeWork_12/Json_hw12", 'w+') as file_json:
```

Comment:
Use the relative path here:
`Json_hw12` instead of `/Users/AleksandrPitelin/PythonProject/PythonProject/HomeWork_12/Json_hw12`
This allows running this code not only on your computer.